examples/milestone/Bank/Bank.juvix

```diff
@@ -0,0 +1,144 @@
+--- This example is taken from Leo workshop:
+---
+--- https://github.com/AleoHQ/workshop/tree/master/basic_bank
+module Bank;
+
+open import Stdlib.Prelude;
+open import Stdlib.Debug.Fail;
+
+open import Stdlib.Data.Nat.Ord;
+
+import Stdlib.Data.Nat as Nat;
+
+Address : Type;
+Address := Nat;
+
+-- TODO replace Nat with Address
+bankAddress : Nat;
+bankAddress := 1234;
+
+--- Some field type.
+axiom Field : Type;
+
+--- Equality test for ;Field;.
+axiom eqField : Field -> Field -> Bool;
+
+module Token;
+  type Token :=
+    | --- Arguments are: owner, gates, amount.
+      mkToken : Address -> Nat -> Nat -> Token;
+
+  --- Retrieves the amount from a ;Token;
+  getOwner : Token -> Address;
+  getOwner (mkToken o _ _) := o;
+
+  --- Retrieves the amount from a ;Token;
+  getAmount : Token -> Nat;
+  getAmount (mkToken _ _ a) := a;
+
+  --- Retrieves the gates from a ;Token;
+  getGates : Token -> Nat;
+  getGates (mkToken _ g _) := g;
+end;
+
+open Token;
+
+--- This module defines the type for balances and its associated operations.
+module Balances;
+  Balances : Type;
+  Balances := List (Field × Nat);
+
+  --- Increments the amount associated with a certain ;Field;.
+  increment : Field -> Nat -> Balances -> Balances;
+  increment f n nil := (f, n) :: nil;
+  increment f n ((b, bn) :: bs) :=
+    if
+      (eqField f b)
+      ((b, bn + n) :: bs)
+      ((b, bn) :: increment f n bs);
+
+  --- Decrements the amount associated with a certain ;Field;.
+  --- If the ;Field; is not found, it does nothing.
+  --- Subtraction is truncated to ;zero;.
+  decrement : Field -> Nat -> Balances -> Balances;
+  decrement _ _ nil := nil;
+  decrement f n ((b, bn) :: bs) :=
+    if
+      (eqField f b)
+      ((b, sub bn n) :: bs)
+      ((b, bn) :: decrement f n bs);
+
+  emtpyBalances : Balances;
+  emtpyBalances := nil;
+
+  --- Commit balances changes to the chain.
+  axiom commitBalances : Balances -> IO;
+end;
+
+open Balances;
+
+--- Runs an ;IO; action on the chain.
+axiom runOnChain : {B : Type} -> IO -> B -> B;
+
+--- Computes the hash associated with an ;Address;.
+axiom hashAddress : Address -> Field;
+
+--- Applies a function n times.
+iterate : {A : Type} -> Nat -> (A -> A) -> A -> A;
+iterate zero _ a := a;
+iterate (suc n) f a := iterate n f (f a);
```

Comment:
What if we actually add this function here: [Stdlib](https://github.com/anoma/juvix-stdlib/tree/main/Stdlib)
/Function.juvix

Reply:
I've created a pr for the stdlib: https://github.com/anoma/juvix-stdlib/pull/52